templates/app/offline-worker.js

```diff
@@ -54,52 +56,44 @@
       var now = Date.now();
       var baseUrl = self.location;
       return this.prepareCache()
-      .then(function (cache) {
-        return Promise.all(this.RESOURCES.map(function (resource) {
-          // Bust the request to get a fresh response
-          var url = new URL(resource, baseUrl);
-          var bustParameter = (url.search ? '&' : '') + '__bust=' + now;
-          var bustedUrl = new URL(url.toString());
-          bustedUrl.search += bustParameter;
-
-          // But cache the response for the original request
-          var requestConfig = { credentials: 'same-origin' };
-          var originalRequest = new Request(url.toString(), requestConfig);
-          var bustedRequest = new Request(bustedUrl.toString(), requestConfig);
-          return fetch(bustedRequest).then(function (response) {
-            if (response.ok) {
-              return cache.put(originalRequest, response);
-            }
-            console.error('Error fetching ' + url + ', status was ' + response.status);
-          });
-        }));
-      }.bind(this));
+      .then(cache => Promise.all(this.RESOURCES.map(resource => {
+        // Bust the request to get a fresh response
+        var url = new URL(resource, baseUrl);
+        var bustParameter = (url.search ? '&' : '') + '__bust=' + now;
+        var bustedUrl = new URL(url.toString());
+        bustedUrl.search += bustParameter;
+
+        // But cache the response for the original request
+        var requestConfig = { credentials: 'same-origin' };
+        var originalRequest = new Request(url.toString(), requestConfig);
+        var bustedRequest = new Request(bustedUrl.toString(), requestConfig);
+        return fetch(bustedRequest)
+        .then(response => {
+          if (response.ok) {
+            return cache.put(originalRequest, response);
+          }
+          console.error('Error fetching ' + url + ', status was ' + response.status);
+        });
+      })));
     },
 
     // Remove the offline caches not controlled by this worker.
     clearOtherCaches: function () {
-      var deleteIfNotCurrent = function (cacheName) {
+      return self.caches.keys()
```

Comment:
I missed the comment. Maybe we can write:
```js
.then(cacheNames => Promise.all(cacheNames.map(cacheName => {
  var isAPreviousCache = cacheName.indexOf(this.CACHE_PREFIX) >= 0 && cacheName !== this.CACHE_NAME;
  return isAPreviousCache ? self.caches.delete(cacheName) : Promise.resolve();
});
```
What do you think (I tend to delay the use of comments as much as possible). You can use `.startsWith()`. It is not ES5 but, you know, this PR is about converting into arrow functions. ;)